zyte_api/__main__.py

```diff
@@ -14,44 +14,62 @@
     create_session,
     AsyncClient,
 )
+from zyte_api.aio.retry import RetryFactory, _is_throttling_error
 from zyte_api.constants import ENV_VARIABLE, API_URL
 from zyte_api.utils import _guess_intype
-from zyte_api.aio.retry import RetryFactory, _is_throttling_error
 
 
 class DontRetryErrorsFactory(RetryFactory):
     retry_condition = retry_if_exception(_is_throttling_error)
 
 
-logger = logging.getLogger('zyte_api')
+logger = logging.getLogger("zyte_api")
 
 _UNSET = object()
 
 
-async def run(queries, out, *, n_conn, stop_on_errors, api_url,
-              api_key=None, retry_errors=True):
+async def run(
+    queries,
+    out,
+    *,
+    n_conn,
+    stop_on_errors,
+    api_url,
+    api_key=None,
+    retry_errors=True,
+    store_errors=None,
+):
+    def write_output(content):
+        json.dump(content, out, ensure_ascii=False)
+        out.write("\n")
+        out.flush()
+        pbar.update()
```

Comment:
TIL An inner function can depend on an outer scope variable as long as it is define before the inner function is *called*, rather than before the inner function is *defined*.